README.md

```diff
@@ -744,6 +745,71 @@ affect your use case.
 
 This can be used for example with PyCharm's [File Watchers](https://www.jetbrains.com/help/pycharm/file-watchers.html).
 
+## blackd
+
+`blackd` is a small HTTP server that exposes *Black*'s functionality over
+a simple protocol. The main benefit of using it is to avoid paying the
+cost of starting up a new *Black* process every time you want to blacken
+a file.
+
+### Usage
+
+You can start the server on the default port, binding only to the local interface
+by running `blackd`. You will see a single line mentioning the server's version,
+and the host and port it's listening on. `blackd` will then print an access log
+similar to most web servers on standard output, merged with any exception traces
+caused by invalid formatting requests.
+
+`blackd` provides even less options than *Black*. You can see them by running
+`blackd --help`:
+
+```text
+Usage: blackd [OPTIONS]
+
+Options:
+  --bind-host TEXT                Address to bind the server to.
+  --bind-port INTEGER             Port to listen on
+  --version                       Show the version and exit.
+  -h, --help                      Show this message and exit.
+```
+
+### Protocol
+
+`blackd` only accepts `POST` requests at the `/` path. The body of the request
+should contain the python source code to be formatted, encoded 
+according to the `charset` field in the `Content-Type` request header. If no
+`charset` is specified, `blackd` assumes `UTF-8`.
+
+There are a few HTTP headers that control how the source is formatted. These
+correspond to command line flags for *Black*. There is one exception to this:
+`X-Protocol-Version` which if present, should have the value `1`, otherwise the
+request is rejected with `HTTP 501` (Not Implemented). 
+
+The headers controlling how code is formatted are:
+
+ - `X-Line-Length`: corresponds to the `--line-length` command line flag.
```

Comment:
Why aren't these just URL parameters, like `POST /?line_length=42&skip_string_normalization=1`? It's going to be a little easier for most clients to set a URL parameter rather than an HTTP header.

Reply:
The reason I chose headers is I think there's less confusion around them compared to POST requests with query parameters. I remember being bitten by frameworks trying to be smart and merging query parameters with the request body.

For most at least somewhat sophisticated HTTP client APIs, AFAIK the only difference between the two approaches is passing a string->string mapping as the `headers` parameter vs `params` in the API call. I don't think this is a big deal.

Reply:
I agree with Zsolt, GET params on POST requests are non-standard. The typical way is to form-encode them and this is admittedly more work for us since we would really need to support multipart/form-data requests at this point.

I agree with Jelle that using headers for this control is kinda weird, too. But this is definitely the easiest thing for us to support on both ends.